invenio_app_ils/vocabularies/sources/base.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Vocabulary base source module."""
+
+from abc import ABCMeta, abstractmethod
+
+from flask import current_app
+from invenio_jsonschemas import current_jsonschemas
+from jsonschema import validate as validate_schema
+
+from invenio_app_ils.errors import VocabularyError
+from invenio_app_ils.records.api import Vocabulary
+
+
+class VocabularySource(metaclass=ABCMeta):
+    """Vocabulary source."""
+
+    def __init__(self, source_type):
+        """Initialize vocabulary source."""
+        self.source_type = source_type
+
+    @staticmethod
+    def validate(vocabulary):
+        """Validate vocabulary."""
+        if not isinstance(vocabulary, Vocabulary):
+            raise VocabularyError(
+                "{!r} is not a vocabulary".format(vocabulary))
```

Comment:
Didn't know about the `!r`! :)

Reply:
Completely useless now since I removed the `__str__` and `__repr__` implementations.. :)